babel/messages/mofile.py

```diff
@@ -7,18 +7,22 @@
     :copyright: (c) 2013-2022 by the Babel Team.
     :license: BSD, see LICENSE for more details.
 """
+from __future__ import annotations
 
 import array
 import struct
+from typing import TYPE_CHECKING
 
 from babel.messages.catalog import Catalog, Message
 
+if TYPE_CHECKING:
+    from _typeshed import SupportsRead, SupportsWrite
```

Comment:
Is this an okay import..? (Ditto for all other `_typeshed` imports)

Reply:
I believe it's a dependency of type-checkers, so as long as `__future__.annotations` is used and it's inside `if TYPE_CHECKING`, it should be fine.

Docs on it are here: https://github.com/python/typeshed/tree/main/stdlib/_typeshed

SupportsItems, SupportsWrite, SupportsRead, and SupportsReadline are marked as stable